### ISSUE TITLE
Getting error while running using the latest storybook version

### ISSUE THREAD

**Issue by shinoy-jose-mindcurv** [kalbeekatz/storybook-a11y-report#32]
I am getting the below error while running  `npx storybook-a11y-report`

![issue](https://user-images.githubusercontent.com/101710233/158570771-2e27b4f2-d5eb-4d24-87b5-0d6a14bd16d5.PNG)

@kalbeekatz

**Comment by JLiu1272** [comment]
Same

**Comment by petetrickey** [comment]
The storybook report is executed against the individual story view, and from what I can tell, window['axe'] is returning as `undefined` because `axe` isn't available from within this screen, only in the main storybook containing view.

* window['axe'] !== undefined (http://localhost:6006/?path=/story/example-button--secondary)
* window['axe'] === undefined (http://localhost:6006/iframe.html?id=example-button--secondary&args=&viewMode=story)

I think injecting axe-core directly as a decorator could solve the issue

**Comment by petetrickey** [comment]
While a longer term fix is outstanding, the following gets it working. Just injecting the bundled axe-core into the preview head file (you might need to create it in your own project).

// ./storybook/preview-head.html
<script src="node_modules/axe-core/axe.min.js"></script>

**Comment by YoungElPaso** [comment]
So the problem is this report depended on Axe from the extension which is no longer loaded the same way presumably?

**Comment by petetrickey** [comment]
Seems like it. The reporter works fine in storybook 6.3, but fails in 6.4 due to the above issue.

**Comment by petetrickey** [comment]
Something like that, yes. Storycrawler is one of the dependancies, which runs puppeteer under the hood. So I guess if you can add a script via that in this library then you would be able to inject axe-core.
On raising a PR, this library has not been updated in over 6 months, so if what you need is time dependant I would probably recommend either using the short term solution above, forking and fixing yourself or potentially going down the Cypress Axe route you mentioned.
Good luck.

**Comment by YoungElPaso** [comment]
Thanks. So I created a patch, which works and I will submit a PR - indeed I just used the Puppeteer API for injecting a script into a page, in this case, axe-core. It's a pretty easy fix and makes the dependency more clear. Hopefully this get's merged soon 🤞  In the meantime, we'll investigate other options.

Luckily, at dev time we already have the Storybook a11y plugin running, so running this in CI is really a nice-to-have double-check, not the first time we'll be looking at a11y stuff.
